File: chapter4/java/src/main/java/com/dotComIt/learnWith/services/AuthenticationService.java
package com.dotComIt.learnWith.services;
import com.dotComIt.learnWith.vos.UserVO;
import java.sql.*;

public class AuthenticationService {
    Connection connection = null;

    public AuthenticationService(Connection connection) {
        this.connection = connection;
    }

    public UserVO authenticate(String username, String password){
        String SQL;
        PreparedStatement pstmt = null;
        ResultSet rs = null;

        try{
            SQL = "SELECT * FROM users where username = ? and password = ?";
            pstmt = connection.prepareStatement(SQL);
            pstmt.setString(1, username);
            pstmt.setString(2, password);
            rs = pstmt.executeQuery();
            if(!rs.isBeforeFirst()){
                return null;
            }  else {
                rs.next();
                UserVO user = new UserVO();
                user.setRoleID(rs.getInt("roleID"));
                user.setUserID(rs.getInt("userID"));
                user.setUsername(rs.getString("userName"));
                return user;
            }
        }
        catch(Exception e){
            e.printStackTrace();
        }
        finally {
            if (rs != null) try { rs.close(); } catch(Exception e) {}
            if (pstmt != null) try { pstmt.close(); } catch(Exception e) {}
            if (connection != null) try { connection.close(); } catch(Exception e) {}
        }
        return null;
    }


}

File: chapter3/angular6App/src/com/dotComIt/learnWith/services/mock/authentication.service.ts
import {UserVO} from "../../vo/UserVO";
import {ResultObjectVO} from "../../vo/ResultObjectVO";

import {Observer} from 'rxjs/index';
import {Observable} from "rxjs/index";

export class AuthenticationService {

    authenticate(username : string, password : string) :
        Observable<ResultObjectVO>  {
        let o : Observable<ResultObjectVO> = Observable.create(
            (observer : Observer<ResultObjectVO>) => {
                setTimeout(() => {
                    let result : ResultObjectVO = new ResultObjectVO();
                    if (( username === 'me' ) && ( password === 'me' )) {
                        result.error = false;
                        result.resultObject = new UserVO();
                        result.resultObject.userID = 1;
                        result.resultObject.username = "me";
                        result.resultObject.roleID = 1;
                    } else if (( username === 'wife' ) && ( password === 'wife' )) {
                        result.error = false;
                        result.resultObject = new UserVO();
                        result.resultObject.userID = 2;
                        result.resultObject.roleID = 2;
                        result.resultObject.username = "wife";
                    } else {
                        result.error = true;
                    }
                    observer.next(result);
                    observer.complete();
                }, 1000);
            });
        return o;
    }

}
